Add required option to config DSL in place of allow blank (#513)
workarea-commerce/workarea: docs/source/articles/configuration-fields.html.md

When defining a configuration field, there are a number of options you can provide.

- `required`

>   Whether or not the field is required, defaults to `false`. Blank values submitted for a field that does not allow it will be set to the field's default value. The admin UI will also prevent user's from submitting blank field values when this is `false`.

Reviewer: This set of sentences is really confusing in the context of `requires` over the old `allow_blank`

```suggestion
  Whether or not the field is required, defaults to `false`. The admin UI will prevent user's from submitting blank field values for required fields. If values are submitted in some other way, blank values for a required field will be set to the field's default value.
```

- `default`

>   The value that will be returned when there isn't a specified value or it is blank. Must match the type defined for the field.

Reviewer: No likey! I don't think an admin setting a non-required field blank is unreasonable.

Author: I'm not so sure, but if you did want to achieve that, you could just not provide a `default` and set `required: false`.
